app/src/main/java/kr/ac/kaist/team888/util/FeatureController.java

```diff
@@ -0,0 +1,73 @@
+package kr.ac.kaist.team888.util;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+
+public class FeatureController {
+  
+  /**
+   * Interface for listening change of hangul features.
+   *
+   * <p> If some feature(weight, curve, and etc)'s value are change,
+   * {@link FeatureController} calls {@link #onFeatureChange()}.
+   *
+   */
+  public interface OnFeatureChangeListener {
+    /**
+     * Method that will be executed on feature's value change.
+     */
+    void onFeatureChange();
+
+    /**
+     * Priority of listener.
+     *
+     * <p> The lower priority executed first.
+     *
+     * @return priority as integer
+     */
+    int getPriority();
+  }
+
+  private static class Singleton {
+    private static final FeatureController instance = new FeatureController();
+  }
+
+  /**
+   * Getter of singleton instance.
+   *
+   * @return singleton instance
+   */
+  public static FeatureController getInstance() {
+    return FeatureController.Singleton.instance;
+  }
+
+  private ArrayList<OnFeatureChangeListener> listeners;
```

Comment:
Any way queue is not appropriate class for iterating, in my opinion.

[PriorityQueue - Oracle](https://docs.oracle.com/javase/7/docs/api/java/util/PriorityQueue.html)
**This class and its iterator implement all of the optional methods of the Collection and Iterator interfaces. The Iterator provided in method iterator() is not guaranteed to traverse the elements of the priority queue in any particular order.**